test/BashProtocol.hs

```diff
@@ -0,0 +1,44 @@
+
+module BashProtocol (testBashProtocol) where
+
+import           Control.Exception
+import           Control.Monad
+import           Data.List
+import           System.Environment
+import           System.FilePath
+import           System.IO
+import           System.IO.Silently
+import           System.Process
+import           Test.Hspec
+
+testBashProtocol :: FilePath -> String -> IO ()
+testBashProtocol program bashProtocol = do
+  assert (takeFileName program == "program") (return ())
+  let protocol = parseProtocol bashProtocol
+  testProtocol (takeDirectory program) protocol
+
+data Protocol
+  = Protocol {
+    _command :: String,
+    _expected :: [String]
```

Comment:
Yeah, it feels very hacky, but given that the bash examples had lots of errors this feels like an important improvement.